src/predictor/gpu_predictor.cu

```diff
@@ -164,7 +169,12 @@ struct DeviceAdapterLoader {
     if (use_shared) {
       return smem[threadIdx.x * columns + fidx];
     }
-    return batch.GetElement(ridx * columns + fidx).value;
+    auto value = batch.GetElement(ridx * columns + fidx).value;
+    if (is_valid(value)) {
+      return nan("");
+    } else {
+      return batch.GetElement(ridx * columns + fidx).value;
```

Comment:
```suggestion
    auto value = batch.GetElement(ridx * columns + fidx).value;
    if (is_valid(value)) {
      return value;
    } else {
      return nan("");
```

If `value` is valid, you shouldn't return a NaN.

Reply:
Thanks for catching that!

Reply:
@trivialfis Can you add a unit test to cover this code?

Reply:
Em .. I need to create a dataset that exceeds the size of shared memory (which is gpu dependent).

Reply:
Let me do that.

Reply:
If it's too difficult, we can merge this PR as it is.

Reply:
Done.  It's good to have you as a reviewer so I can try harder on testing.

Reply:
Good to hear that my review helped.